src/accessibility/browser.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  getRole,
+  computeAccessibleName,
+  computeAccessibleDescription,
+} from 'dom-accessibility-api';
+import type { AccessibilityTreeOptions } from '.';
+
+const indent = (text: string, indenter = '  ') =>
+  indenter + text.split('\n').join(`\n${indenter}`);
+
+const enum AccessibilityState {
+  /** This element is accessible and its descendents could be accessible */
+  Accessible,
+  /** This element is innaccessible, but its descendents could be accessible */
+  InaccessibleSelf,
+  /** Both this element and its descendents are inaccessible */
+  InaccessibleSelfAndDescendents,
+}
+
+// TODO in PR: what about role="presentation" or role="none"? Should they be excluded?
+// The spec says "it depends on the implementation" basically I think
+const getAccessibilityState = (element: Element): AccessibilityState => {
+  const computedStyle = getComputedStyle(element);
+  if (
+    (element as HTMLElement).hidden ||
+    element.getAttribute('aria-hidden') === 'true' ||
+    computedStyle.display === 'none'
+  )
+    return AccessibilityState.InaccessibleSelfAndDescendents;
+
+  // An element can have visibility: 'hidden' but its descendents can override visibility
+  if (computedStyle.visibility === 'hidden')
+    return AccessibilityState.InaccessibleSelf;
+
+  return AccessibilityState.Accessible;
+};
+
+export const getAccessibilityTree = (
+  element: Element,
+  opts: AccessibilityTreeOptions,
+): string => {
+  const accessibilityState = getAccessibilityState(element);
+  if (accessibilityState === AccessibilityState.InaccessibleSelfAndDescendents)
+    return '';
+  const { includeDescriptions = true, includeText = false } = opts;
```

Comment:
TIL you can use object destructuring with default values.

Reply:
Yep, it's pretty handy!